docs/docs-beta/docs/dagster-plus/deployment/branch-deployments/setting-up-branch-deployments.md

```diff
@@ -0,0 +1,484 @@
+       module: dagster_cloud.workspace.docker
+       class: DockerUserCodeLauncher
+       config:
+         networks:
+           - dagster_cloud_agent
+         server_ttl:
+           enabled: true
+           ttl_seconds: 7200 #2 hours
+     ```
+
+  </TabItem>
+  <TabItem value="k8s" label="Kubernetes" default>
+
+  1. Set up a new Kubernetes agent. Refer to the [Kubernetes agent setup guide](/dagster-plus/deployment/hybrid/agents/kubernetes) for instructions.
+
+  2. After the agent is set up, modify your Helm values file to include the following:
+
+     ```yaml
+     dagsterCloud:
+       branchDeployments: true
+     ---
+     workspace:
+       serverTTL:
+         enabled: true
+         ttlSeconds: 7200
+     ```
+
+  </TabItem>
+</Tabs>
+
+## Step 4: Set up branch  deployments
+
+<Tabs groupId="method">
+  <TabItem value="github" label="GitHub">
+
+### Step 4.1: Add GitHub CI/CD script to your project
+  :::note
+If you used the GitHub app to add the project as a code location, this step isn't required. [Skip ahead](#step-42-add-the-agent-registry-to-dagster_cloudyaml)
+  :::
+
+Copy the following files to your project:
+
+- [`dagster_cloud.yaml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/dagster_cloud.yaml)
+- [`.github/workflows/dagster-cloud-deploy.yml`](https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/.github/workflows/dagster-cloud-deploy.yml) (for **Hybrid** deployments)
+- [`.github/workflows/branch_deployments.yml`](https://github.com/dagster-io/dagster-cloud-serverless-quickstart/blob/main/.github/workflows/branch_deployments.yml) (for **Serverless** deployments)
+
+In the next step, you'll modify these files to work with your Dagster+ setup.
+
+### Step 4.2: Add the agent registry to dagster_cloud.yaml
+
+:::note
+If you're using a Serverless deployment, this step isn't required. [Skip ahead](#step-45-verify-github-action-runs)
+:::
+
+In the `dagster_cloud.yaml` file, replace `build.registry` with the registry used by the [agent you created in Step 2](#step-2-generate-a-dagster-agent-token).
+
+For example:
+
+<CodeExample filePath="dagster-plus/deployment/branch-deployments/dagster_cloud.yaml" language="yaml" />
+
+### Step 4.3: Configure GitHub Action secrets
+
+:::note
+If you're using a Serverless deployment, this step isn't required. [Skip ahead](#step-45-verify-github-action-runs)
+:::
+:::note
+If you used the GitHub app to add the project as a code location, this step isn't required. [Skip ahead](#step-44-configure-github-action).
+:::
+
+1. In the repository, click the **Settings** tab.
+2. In the **Security** section of the sidebar, click **Secrets > Actions**.
+3. Click **New repository secret**.
+4. In the **Name** field, enter the name of the secret. For example, `DAGSTER_CLOUD_API_TOKEN`
+5. In the **Value** field, paste the value of the secret.
+6. Click **Add secret**.
+
+Repeat steps 3-6 for each of the secrets required for your registry type:
```

Comment:
OH I see that the info is there in the tabs. Leaving my original comment though because i think it is still indicative that the tabs are not obviously part of this step. Maybe it's because they look like a heading?

Reply:
It still could be nice to tell the user what info to collect while they are making the agent so that they dont have to go back and find it during this step